streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -246,6 +252,7 @@ void handleTaskLoss() {
             // standby tasks while we rejoin.
             if (task.isActive()) {
                 task.closeDirty();
+                changelogReader.remove(task.changelogPartitions());
```

Comment:
Whenever we are closing a task (clean or dirty), we should call changelog reader to remove the partitions from its managed restore consumer.